runtime-modules/content/src/nft/types.rs

```diff
@@ -0,0 +1,329 @@
+    }
+
+    /// Create new NFT
+    pub fn new(owner: NFTOwner<MemberId>, creator_royalty: Option<Royalty>) -> Self {
+        Self {
+            owner,
+            transactional_status: TransactionalStatus::Idle,
+            creator_royalty,
+        }
+    }
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum NFTOwner<MemberId> {
+    ChannelOwner,
+    Member(MemberId),
+}
+
+impl<MemberId> Default for NFTOwner<MemberId> {
+    fn default() -> Self {
+        Self::ChannelOwner
+    }
+}
+
+/// Information on the auction being created.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct Bid<MemberId, BlockNumber: BaseArithmetic + Copy, Balance> {
+    pub bidder: MemberId,
+    pub amount: Balance,
+    pub time: BlockNumber,
+}
+
+impl<MemberId, BlockNumber: BaseArithmetic + Copy, Balance> Bid<MemberId, BlockNumber, Balance> {
+    fn new(bidder: MemberId, amount: Balance, time: BlockNumber) -> Self {
+        Self {
+            bidder,
+            amount,
+            time,
+        }
+    }
+}
+
+/// Information on the auction being created.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct AuctionRecord<BlockNumber: BaseArithmetic + Copy, Balance, MemberId: Ord> {
+    pub starting_price: Balance,
+    pub buy_now_price: Option<Balance>,
+    /// Auction type (either english or open)
+    pub auction_type: AuctionType<BlockNumber>,
+    pub minimal_bid_step: Balance,
+    pub last_bid: Option<Bid<MemberId, BlockNumber, Balance>>,
+    pub starts_at: Option<BlockNumber>,
+    pub whitelist: Option<BTreeSet<MemberId>>,
+}
+
+impl<
+        BlockNumber: BaseArithmetic + Copy + Default,
+        Balance: Default + BaseArithmetic,
+        MemberId: Default + PartialEq + Ord,
+    > AuctionRecord<BlockNumber, Balance, MemberId>
+{
+    /// Create a new auction record with provided parameters
+    pub fn new<VideoId>(
+        auction_params: AuctionParams<VideoId, BlockNumber, Balance, MemberId>,
+    ) -> Self {
+        let AuctionParams {
+            auction_type,
+            starting_price,
+            buy_now_price,
+            minimal_bid_step,
+            starts_at,
+            whitelist,
+            ..
+        } = auction_params;
+        Self {
+            starting_price,
+            buy_now_price,
+            auction_type,
+            minimal_bid_step,
+            last_bid: None,
+            starts_at,
+            whitelist,
+        }
+    }
+
+    /// Ensure new bid is greater then last bid + minimal bid step
+    pub fn ensure_is_valid_bid<T: Trait>(&self, new_bid: Balance) -> DispatchResult {
+        // Always allow to buy now
+        match &self.buy_now_price {
+            Some(buy_now_price) if new_bid >= *buy_now_price => (),
```

Comment:
> I don't think you need to check for this explicitly. Whether you are less or more than any possibly set buy now price is not relevant on its own.

The reason i did this check, because i want always to allow `buy_now` even though `minimal_bid_step` delta not satisfied, for example.

Reply:
Ah, of course, very nice catch.